File: 0404.左叶子之和/0404-左叶子之和.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */


int sumOfLeftLeaves(struct TreeNode* root){
    if (root == NULL) {
        return 0;
    }
    if (root->left != NULL && root->left->left == NULL && root->left->right == NULL) {
        return root->left->val + sumOfLeftLeaves(root->right); 
    } 
    return sumOfLeftLeaves(root->left) + sumOfLeftLeaves(root->right);
}

File: 0404.左叶子之和/0404-左叶子之和.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def sumOfLeftLeaves(self, root: TreeNode) -> int:
        def h(node):
            if not node:
                return 0
            if node.left and not node.left.left and not node.left.right:
                return node.left.val + h(node.right)
            return h(node.left) + h(node.right)
        return h(root)
            